### PR TITLE
update the latest ndv5 algo optimization

### PR DESCRIPTION
**Pull request by Andyli1007**
update part including
1. README with latest perf numbers
2. doc about how to repro the perf numbers
3. latest commit related to the algo files changes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,6 +14,236 @@ MSCCL vision is to provide a unified, efficient, and scalable framework for exec
 
 - MSCCL test toolkit([msccl-tests-nccl](https://github.com/Azure/msccl-tests-nccl)): These tests check both the performance and the correctness of MSCCL operations.
 
+## Performance
+For reference, FP16 All-Reduce and All-Gather algorithms were tested and compared on a VM running Ubuntu 20.04 (Linux 5.11.0-41-generic), with 8 Nvidia H100 80G GPU, using msccl-tests-nccl.
```
Comment (yzygitzh) [inline README.md:18]: We can directly say it's a ND H100 v5 VM

File: docs/performance-ndmv4.md [hunk 2]
```diff
@@ -0,0 +1,236 @@
+# NDmv4 Performance
+
+All results from NDmv4. NCCL version 2.18.1+cuda12.1, reported in-place numbers.
```
Comment (yzygitzh) [inline docs/performance-ndmv4.md:3]: It's ND H100 v5 instead of NDmv4?

Comment (yzygitzh) [reply docs/performance-ndmv4.md:3]: NCCL version 2.18.3

File: docs/performance-ndmv4.md [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+
+nccl-tests command example:
+```bash
+mpirun --allow-run-as-root --tag-output -map-by ppr:8:node --bind-to numa -mca pml ob1 -mca btl ^openib -mca btl_tcp_if_include eth0 -x PATH -x LD_PRELOAD=/path/to/msccl-executor-nccl/build/lib/libnccl.so -x NCCL_IB_PCI_RELAXED_ORDERING=1 -x NCCL_SOCKET_IFNAME=eth0 -x CUDA_DEVICE_ORDER=PCI_BUS_ID -x NCCL_TOPO_FILE=/path/to/ndv5-topo.xml -x NCCL_DEBUG=WARN -x NCCL_MIN_NCHANNELS=32 /path/to/msccl-tests-nccl/build/all_gather_perf -b 1 -e 1G -f 2 -g 1 -c 1 -w 20 -n 1000 -d half -G 1
+```
```
Comment (yzygitzh) [inline docs/performance-ndmv4.md:8]: Add all-reduce command here as well

File: docs/performance-ndmv4.md [hunk 4]
```diff
@@ -0,0 +1,236 @@
+# NDmv4 Performance
+
+All results from NDmv4. NCCL version 2.18.1+cuda12.1, reported in-place numbers.
+
+nccl-tests command example:
```
Comment (yzygitzh) [inline docs/performance-ndmv4.md:5]: msccl-tests-nccl

File: docs/performance-ndmv4.md (hunk 2 above)
Comment (yzygitzh) [reply docs/performance-ndmv4.md:3]: The file name should be performance-nd-h100-v5.md

File: README.md (hunk 1 above)
Comment (Andyli1007) [reply README.md]: updated

File: docs/performance-ndmv4.md (hunk 3 above)
Comment (Andyli1007) [reply docs/performance-ndmv4.md:9]: updated

File: docs/performance-ndmv4.md (hunk 4 above)
Comment (Andyli1007) [reply docs/performance-ndmv4.md]: updated

**Review by yzygitzh — APPROVED** [review]

**Pull request merged**